Graphics/GraphicsEngine/interface/GraphicsTypes.h

```diff
@@ -1895,12 +1895,20 @@ struct Version
     constexpr Version() noexcept
     {}
 
+#if !defined(DILIGENT_SHARP_GEN)
     template <typename T, typename = typename std::enable_if<std::is_integral<T>::value>::type>
     constexpr Version(T _Major, T _Minor) noexcept :
         Major{static_cast<decltype(Major)>(_Major)},
         Minor{static_cast<decltype(Minor)>(_Minor)}
     {
     }
+#else
+    constexpr Version(Uint32 _Major, Uint32 _Minor) noexcept :
```

Comment:
Is template the problem here?

Reply:
CastXML uses clang 13.0.0 for parsing C++ code. The latest C++ headers in MSVC require clang version 15.0.0
https://github.com/CastXML/CastXMLSuperbuild/issues/53